Add development Dockerfile

CannonLock opened the pull request:
99% of the work is courtesy @jhiemstrawisc and his Dockerfile.

Draft for now well I figure out why I can't run Goreleaser, everything else works as expected. 

Closes #88.

jhiemstrawisc commented on .github/workflows/publish-dev-container.yml:
```diff
@@ -0,0 +1,48 @@
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: hub.opensciencegrid.org/macrostrat/tileserver
```
Do we mean to be grabbing things from the macrostrate container, or is this a copy-paste error?

jhiemstrawisc commented on images/dev.Dockerfile:
```diff
@@ -0,0 +1,39 @@
+#FROM mcr.microsoft.com/devcontainers/go:0-1-bullseye
```
Let's remove this comment

jhiemstrawisc commented on images/dev.Dockerfile:
```diff
@@ -0,0 +1,39 @@
+#FROM mcr.microsoft.com/devcontainers/go:0-1-bullseye
+FROM almalinux:8
```
We should pin the platform as arm here, otherwise this Dockerfile (which is tailored for arm) will be built using the GHA runner's host OS -- most definitely amd64.

jhiemstrawisc commented on images/dev.Dockerfile:
```diff
@@ -0,0 +1,39 @@
+
+# Installing the right version of go
+SHELL ["/bin/sh", "-c"]
+RUN curl https://dl.google.com/go/go1.20.8.linux-arm64.tar.gz -o go1.20.8.linux-arm64.tar.gz 
```
As it is, this only builds a container for arm64 -- great for us Mac users, but not useful for anyone else. Can we make two dev containers, one for amd64 and one for arm64? The amd64 container should use the OSG base containers. I couldn't use those for arm because the OSG repositories don't package xrootd for arm.

In the amd64 container, this line will need to be modified to grab the correct arch

jhiemstrawisc reviewed (changes requested):
The big things are that we should make sure a container is built for archs other than arm (as much as I'd love to flip the script on all those folks who don't build for arm64!)

CannonLock commented:
@jhiemstrawisc This addresses your comments. I am going to refrain from creating an amd64 image from the osg-software base until a need is identified. I think it makes sense to have us all use the same dev image across both platforms.

jhiemstrawisc reviewed (approved):
LGTM
